homeassistant/components/sensor/__init__.py

```diff
@@ -95,21 +96,43 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return await component.async_unload_entry(entry)
 
 
+@dataclass
+class SensorEntityDescription(EntityDescription):
+    """An class that describes sensor entities."""
+
+    state_class: str | None = None
+    last_reset: datetime | None = None
+
+
 class SensorEntity(Entity):
     """Base class for sensor entities."""
 
-    _attr_state_class: str | None = None
-    _attr_last_reset: datetime | None = None
+    _attr_entity_description: SensorEntityDescription
+    _attr_state_class: str | None
+    _attr_last_reset: datetime | None
+
+    @property
+    def entity_description(self) -> SensorEntityDescription:
+        """Return the entity description of this entity."""
+        return self._attr_entity_description
 
     @property
     def state_class(self) -> str | None:
         """Return the state class of this entity, from STATE_CLASSES, if any."""
-        return self._attr_state_class
+        if hasattr(self, "_attr_state_class"):
+            return self._attr_state_class
+        if self.entity_description:
```

Comment:
Should this be a `hasattr` check too? Now this will raise `AttributeError`

Reply:
Uh, yeah now it does... well spotted.